lib/internal/streams/pipeline.js

```diff
@@ -189,8 +221,15 @@ function pipeline(...streams) {
       return;
     }
 
-    while (destroys.length) {
-      destroys.shift()(error);
+    while (
+      (hasTransform && destroys.length - 1) ||
+      (!hasTransform && destroys.length)
+    ) {
+      try {
+        destroys.shift()(error);
+      } catch {
+        // Untreated destroys error
```

Comment:
What would happen here?

Reply:
In some cases the shift function are not encountered